libraries/AP_Compass/CompassCalibrator.cpp

```diff
@@ -768,6 +768,23 @@ Vector3f CompassCalibrator::calculate_earth_field(CompassSample &sample, enum Ro
     return efield;
 }
 
+/*
+  return true if two rotations are equivalent
+  This copes with the fact that we have some duplicates, like ROLL_180_YAW_90 and PITCH_180_YAW_270
+ */
+bool CompassCalibrator::rotation_equal(enum Rotation r1, enum Rotation r2) const
+{
+    if (r1 == r2) {
+        return true;
+    }
+    Vector3f v(1,2,3);
```

Comment:
It might be worth spacing this out more, there is a paranoid voice in my head that is wondering if there is a rotation set that can make `v(3, 2, 1)`